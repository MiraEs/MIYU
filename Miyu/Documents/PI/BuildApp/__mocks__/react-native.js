/* eslint-disable react/no-multi-comp */
import {
	createElement,
	Component,
} from 'react';
import PropTypes from 'prop-types';
const ReactNative = jest.genMockFromModule('react-native');

ReactNative.StyleSheet = {
	create: function create(styles) {
		return styles;
	},
	flatten: function flatten(...styles) {
		return Object.assign({}, ...styles);
	},
};

ReactNative.PixelRatio = {
	get() {
		return 1;
	},
};

ReactNative.Alert = {
	alert: jest.fn(),
};

ReactNative.StatusBar = {
	setBarStyle: jest.fn(),
	setHidden: jest.fn(),
};

ReactNative.Linking = {
	openURL: jest.fn(),
};

ReactNative.NativeModules = {
	PhotoUploader: {},
	SafariViewManager: {},
	UIManager: {
		measure: jest.fn(),
	},
	MarketingCloud: {
		trackAction: jest.fn(),
		trackState: jest.fn(),
	},
	LocalyticsManager: {
		tagScreen: jest.fn(),
		tagEvent: jest.fn(),
		tagCustomerLoggedOut: jest.fn(),
	},
	Instabug: {
		log: jest.fn(),
	},
	BNDevice: {},
	RNApplePayManager: {
		canMakePayments: jest.fn(() => Promise.resolve(true)),
		canMakePaymentsUsingNetworks: jest.fn(() => Promise.resolve(true)),
		paymentRequest: jest.fn(),
		paymentRequestDidFinish: jest.fn(),
		updateShippingMethods: jest.fn(),
		selectShippingMethod: jest.fn(),
		authorizedPayment: jest.fn(),
	},
};

ReactNative.requireNativeComponent = jest.fn();

ReactNative.NetInfo = {
	isConnected: {
		fetch: jest.fn(() => ({ done: jest.fn() })),
		addEventListener: jest.fn(),
		removeEventListener: jest.fn(),
	},
};

ReactNative.AppState = {
	addEventListener: jest.fn(),
};

ReactNative.CameraRoll = {
	getPhotos: jest.fn(() => Promise.resolve([])),
};

ReactNative.PushNotificationIOS = {
	setApplicationIconBadgeNumber: jest.fn(),
};

ReactNative.Keyboard = {
	dismiss: jest.fn(),
};

ReactNative.findNodeHandle = jest.fn();

ReactNative.ViewPropTypes = {
	style: jest.fn(),
};

ReactNative.NativeEventEmitter = class NativeEventEmitter {
	addListener = jest.fn(() => ({ remove: jest.fn() }));
};

class FlatList extends Component {
	render() {
		return createElement('FlatList', this.props);
	}
}

class SectionList extends Component {
	render() {
		return createElement('SectionList', this.props);
	}
}

class View extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('View', this.props, this.props.children);
	}
}

class ScrollView extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('ScrollView', this.props, this.props.children);
	}
}

class Switch extends Component {
	render() {
		return createElement('Switch');
	}
}

class Text extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('Text', this.props, this.props.children);
	}
}

class TextInput extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TextInput', this.props, this.props.children);
	}
}

class TouchableOpacity extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TouchableOpacity', this.props, this.props.children);
	}
}

class TouchableWithoutFeedback extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TouchableWithoutFeedback', this.props, this.props.children);
	}
}

class ToolbarAndroid extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('ToolbarAndroid', this.props, this.props.children);
	}
}

class Image extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('Image', this.props, this.props.children);
	}
}

class Icon extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('Icon', this.props, this.props.children);
	}
}

class ActivityIndicator extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('ActivityIndicator', this.props, this.props.children);
	}
}

ReactNative.Platform = {
	OS: 'ios',
	select: jest.fn((options) => options.ios),
};

ReactNative.InteractionManager = {
	runAfterInteractions: jest.fn(),
};

ReactNative.ActionSheetIOS = {
	showActionSheetWithOptions: jest.fn(),
};

class ListView extends Component {
	DataSource() {
		return {
			cloneWithRows: jest.fn(),
			cloneWithRowsAndSections: jest.fn(),
		};
	}
}

class AppRegistry {
	registerComponent() {}
}

class TouchableHighlight extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TouchableHighlight', this.props, this.props.children);
	}
}
class BackHandler extends Component {
	static addEventListener() {}
	static exitApp() {}
	static removeEventListener() {}
}

class TouchableNativeFeedback extends Component {
	static Ripple() {
	}
}

class Navigator extends Component {
	render() {
	}
}

class RefreshControl extends Component {
	render() {
		return createElement('RefreshControl', this.props);
	}
}

class LayoutAnimation extends Component {
	static configureNext() {
	};

	static Presets() {
		return {
			easeInEaseOut: jest.fn(),
		};
	};
}

class Dimensions {
	static get() {
		return 0;
	}
}

class Animated extends Component {

	static Value() {
		return {
			getLayout: () => {},
			interpolate: () => {},
			setOffset: () => {},
			setValue: () => {},
		};
	}

	static ValueXY() {
		return {
			getLayout: () => {},
			interpolate: () => {},
			setOffset: () => {},
			setValue: () => {},
		};
	}

	static event() {}
}

class AnimatedView extends Component {
	render() {
		return createElement('AnimatedView');
	}
}

class AnimatedImage extends Component {

}

class AnimatedText extends Component {

}

Animated.View = AnimatedView;

Animated.Image = AnimatedImage;

Animated.Text = AnimatedText;

class PanResponder {
	static create() {
		return {panHandlers: {}};
	}
}

global.__fbBatchedBridgeConfig = {
	remoteModuleConfig: [],
	localModulesConfig: [],
};

ReactNative.View = View;
ReactNative.ScrollView = ScrollView;
ReactNative.FlatList = FlatList;
ReactNative.ListView = ListView;
ReactNative.Text = Text;
ReactNative.TextInput = TextInput;
ReactNative.TouchableOpacity = TouchableOpacity;
ReactNative.TouchableHighlight = TouchableHighlight;
ReactNative.TouchableWithoutFeedback = TouchableWithoutFeedback;
ReactNative.ToolbarAndroid = ToolbarAndroid;
ReactNative.Image = Image;
ReactNative.AppRegistry = AppRegistry;
ReactNative.Icon = Icon;
ReactNative.ActivityIndicator = ActivityIndicator;
ReactNative.BackHandler = BackHandler;
ReactNative.TouchableNativeFeedback = TouchableNativeFeedback;
ReactNative.Navigator = Navigator;
ReactNative.RefreshControl = RefreshControl;
ReactNative.LayoutAnimation = LayoutAnimation;
ReactNative.Dimensions = Dimensions;
ReactNative.Animated = Animated;
ReactNative.PanResponder = PanResponder;
ReactNative.SectionList = SectionList;
ReactNative.Switch = Switch;

module.exports = ReactNative;
