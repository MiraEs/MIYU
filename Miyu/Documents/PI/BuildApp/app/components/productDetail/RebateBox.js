'use strict';

import React, {
	Component,
} from 'react';
import PropTypes from 'prop-types';
import {
	StyleSheet,
	View,
} from 'react-native';
import {
	LinkButton,
	Text,
} from 'BuildLibrary';
import styles from '../../lib/styles';
import helpers from '../../lib/helpers';
import helpersWithLoadRequirements from '../../lib/helpersWithLoadRequirements';
import environment from '../../lib/environment';
import Icon from 'react-native-vector-icons/Ionicons';
import pluralize from 'pluralize';
import TrackingActions from '../../lib/analytics/TrackingActions';

const componentStyles = StyleSheet.create({
	rebateBox: {
		paddingHorizontal: styles.measurements.gridSpace1,
		paddingVertical: styles.measurements.gridSpace2,
		marginBottom: styles.measurements.gridSpace2,
		flexDirection: 'row',
		backgroundColor: 'white',
	},
	textColumn: {
		marginLeft: styles.measurements.gridSpace2,
		flex: 1,
	},
	bulletRowWrapper: {
		marginTop: styles.measurements.gridSpace1,
		flexDirection: 'row',
		paddingLeft: styles.measurements.gridSpace2,
	},
	bulletText: {
		flex: 1,
		marginLeft: styles.measurements.gridSpace3,
	},
	detailsContainer: {
		marginBottom: styles.measurements.gridSpace2,
	},
});

class RebateBox extends Component {
	constructor(props) {
		super(props);

		this.state = {
			showDetails: false,
		};
	}

	getDetails = () => {
		const {
			rebates,
			rebates: { length },
		} = this.props;

		if (this.state.showDetails) {
			return (
				<View style={componentStyles.detailsContainer}>
					<Text>
						{length > 1 ? 'These' : 'A'} mail-in {pluralize('rebate', length, false)}{' '}
						{pluralize('is', length, false)} available for this product!
					</Text>
					{
						rebates.map((rebate) => {
							return (
								<View key={rebate.rebateId}>
									<View style={componentStyles.bulletRowWrapper}>
										<Text>&bull;</Text>
										<View style={componentStyles.bulletText}>
											<LinkButton onPress={() => helpersWithLoadRequirements.openURL(`${environment.secureHost}${rebate.link}`, TrackingActions.REBATE_LINK)}>
												<Text color="primary">{rebate.linkText}</Text>
											</LinkButton>
											<Text
												color="greyDark"
												size="small"
												style={componentStyles.offerEnds}
											>
												Offer Ends {helpers.getDateStrictFormat(rebate.endDate)}.
												Send By {helpers.getDateStrictFormat(rebate.sendByDate)}
											</Text>
										</View>
									</View>
								</View>
							);
						})
					}
				</View>
			);
		}
	};

	getShowHideText = () => {
		return this.state.showDetails ? 'Hide' : 'View';
	};

	render() {
		const { rebates: { length } } = this.props;

		return (
			<View style={componentStyles.rebateBox}>
				<Icon
					color={styles.colors.accent}
					name="md-mail-open"
					size={30}
				/>
				<View style={componentStyles.textColumn}>
					<Text
						color="accent"
						weight="bold"
					>
						{pluralize('Rebate', length, false)} Available
					</Text>
					{this.getDetails()}
					<View style={styles.elements.flexRow}>
						<LinkButton onPress={() => this.setState({ showDetails: !this.state.showDetails })}>
							<Text
								color="primary"
								size="small"
								lineHeight={false}
							>
								{this.getShowHideText()} Details
							</Text>
						</LinkButton>
					</View>
				</View>
			</View>
		);
	}
}

RebateBox.propTypes = {
	rebates: PropTypes.arrayOf(
		PropTypes.shape({
			link: PropTypes.string.isRequired,
			linkText: PropTypes.string.isRequired,
			endDate: PropTypes.oneOfType([
				PropTypes.string,
				PropTypes.number,
			]).isRequired,
			sendByDate: PropTypes.oneOfType([
				PropTypes.string,
				PropTypes.number,
			]).isRequired,
			startDate: PropTypes.oneOfType([
				PropTypes.string,
				PropTypes.number,
			]).isRequired,
		}),
	),
};

RebateBox.defaultProps = {
	rebates: [],
};

export default RebateBox;
