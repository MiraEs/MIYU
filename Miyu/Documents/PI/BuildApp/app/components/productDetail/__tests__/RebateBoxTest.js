jest.unmock('react-native');
jest.mock('../../../services/httpClient', () => ({}));
jest.mock('../../../lib/styles');
import 'react-native';
import React from 'react';
import RebateBox from '../RebateBox';

jest.mock('BuildLibrary');
jest.mock('BuildNative');

const oneRebate = {
	rebates: [{
		title: 'Test Sale',
		description: 'Test sale details.',
		active: true,
	}],
};
const manyRebates = {
	rebates: [{
		rebateId: 32,
		startDate: 1360828800000,
		endDate: 1364367600000,
		sendByDate: 1366873200000,
		link: 'http://products.geappliances.com/MarketingObjectRetrieval/Dispatcher?RequestType=PDF&Name=72807_q1_abd_cafe_rebate.pdf',
		linkText: 'Get up $500 back with select GE Cafe items!',
	}, {
		rebateId: 52,
		startDate: 1367478000000,
		endDate: 1373439600000,
		sendByDate: 1375945200000,
		link: 'http://products.geappliances.com/MarketingObjectRetrieval/Dispatcher?RequestType=PDF&Name=74594A_q2_cafe_web_reb.pdf',
		linkText: 'Save up to $750 with 50% bonus savings',
	}],
};
const noRebates = {
	rebates: [],
};


describe('RebateBox Container', () => {
	it('should render with one rebate', () => {
		const tree = require('react-test-renderer').create(
			<RebateBox {...oneRebate} />
		);
		expect(tree.toJSON()).toMatchSnapshot();
	});

	it('should render with more than one rebate', () => {
		const tree = require('react-test-renderer').create(
			<RebateBox {...manyRebates} />
		);
		expect(tree.toJSON()).toMatchSnapshot();
	});

	it('should render with no rebates', () => {
		const tree = require('react-test-renderer').create(
			<RebateBox {...noRebates} />
		);
		expect(tree.toJSON()).toMatchSnapshot();
	});
});
