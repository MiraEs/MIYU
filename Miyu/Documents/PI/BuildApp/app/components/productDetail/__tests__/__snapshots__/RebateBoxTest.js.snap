// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RebateBox Container should render with more than one rebate 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flexDirection": "row",
      "marginBottom": 16,
      "paddingHorizontal": 8,
      "paddingVertical": 16,
    }
  }
>
  <Icon
    color="#D45D00"
    name="md-mail-open"
    size={30}
  />
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 16,
      }
    }
  >
    <Text
      color="accent"
      weight="bold"
    >
      Rebates
       Available
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <LinkButton
        onPress={[Function]}
      >
        <Text
          color="primary"
          lineHeight={false}
          size="small"
        >
          View
           Details
        </Text>
      </LinkButton>
    </View>
  </View>
</View>
`;

exports[`RebateBox Container should render with no rebates 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flexDirection": "row",
      "marginBottom": 16,
      "paddingHorizontal": 8,
      "paddingVertical": 16,
    }
  }
>
  <Icon
    color="#D45D00"
    name="md-mail-open"
    size={30}
  />
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 16,
      }
    }
  >
    <Text
      color="accent"
      weight="bold"
    >
      Rebates
       Available
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <LinkButton
        onPress={[Function]}
      >
        <Text
          color="primary"
          lineHeight={false}
          size="small"
        >
          View
           Details
        </Text>
      </LinkButton>
    </View>
  </View>
</View>
`;

exports[`RebateBox Container should render with one rebate 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flexDirection": "row",
      "marginBottom": 16,
      "paddingHorizontal": 8,
      "paddingVertical": 16,
    }
  }
>
  <Icon
    color="#D45D00"
    name="md-mail-open"
    size={30}
  />
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 16,
      }
    }
  >
    <Text
      color="accent"
      weight="bold"
    >
      Rebate
       Available
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <LinkButton
        onPress={[Function]}
      >
        <Text
          color="primary"
          lineHeight={false}
          size="small"
        >
          View
           Details
        </Text>
      </LinkButton>
    </View>
  </View>
</View>
`;
